src/network/clientpackethandler.cpp

```diff
@@ -1161,6 +1161,23 @@ void Client::handleCommand_HudSetSky(NetworkPacket* pkt)
 	m_client_event_queue.push(event);
 }
 
+void Client::handleCommand_HudSetClouds(NetworkPacket* pkt)
+{
+	std::string datastring(pkt->getString(0), pkt->getSize());
+	std::istringstream is(datastring, std::ios_base::binary);
+
+	u16 density                      = readU16(is);
+	video::SColor *color             = new video::SColor(readARGB8(is));
```

Comment:
This is a memory leak as there is no corresponding delete.

I don't think the color member of set_clouds needs to be a pointer at all, because SColor is very cheap to copy (basically a 32 bit integer).